optimism/src/keccak/constraints.rs

```diff
@@ -246,7 +246,7 @@ impl<Fp: Field> Constraints for KeccakEnv<Fp> {
             // STEP iota: 4 constraints
             for (q, c) in self.round_constants().iter().enumerate() {
                 self.constrain(
-                    self.is_round() * (self.next_state(q) - (state_f[0][0][q].clone() + c.clone())),
+                    self.is_round() * (self.state_g(q) - (state_f[0][0][q].clone() + c.clone())),
```

Comment:
Is it OK that you only constrain `state_g` here once`? My very vague intuition suggests that that it should appear in two places, similarly to `new_state(i)`...?

Reply:
For transparency, in our meeting we discussed how the step in the interpreter should take the value of `state_g` to instantiate the `prev_step` of a round. But now that I am checking the code, I can see that `prev_step` is indeed used to initialize sponges and for the first theta, but it is not done in-between rounds. Will be creating upcoming PR to address this issue.